core/src/test/java/org/apache/calcite/test/ScannableTableTest.java

```diff
@@ -155,7 +155,7 @@
             "j=Paul");
     // Only 2 rows came out of the table. If the value is 4, it means that the
     // planner did not pass the filter down.
-    assertThat(buf.toString(), is("returnCount=2, filter=<0, 4>, projects=[1]"));
+    assertThat(buf.toString(), is("returnCount=2, filter=<0, 4>, projects=[1, 0]"));
```

Comment:
Is it expected to have this side effect (and the one in line 191)?

Reply:
@rubenada Yes, it's expected. The test ignored this case.

Reply:
@IceMimosa Do you know what projetcs means?

Reply:
@JiajunBernoulli This test wants to pushdown filter to table scan, but it's not suitable here, it should use RelRule to do this in optimize phase. `TableScan's projects` and `ProjectableFilterableTable's projects` are not the same, The test here is `select j where i`, in physical execution, the ProjectableFilterableTable's projects must contain `j` and `i` if you don't change the AST.